src/compiler/checker.ts

```diff
@@ -18056,7 +18278,30 @@ namespace ts {
         }
 
         function getLengthOfTupleType(type: TupleTypeReference) {
-            return getTypeReferenceArity(type) - (type.target.hasRestElement ? 1 : 0);
+            return type.target.fixedLength;

```

Comment:
Seems like this function should either be inlined and removed, or other places where `type.target.fixedLength` is accessed should call it instead (e.g. line 18272 above)

Reply:
Agree.